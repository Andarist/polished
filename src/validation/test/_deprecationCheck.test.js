// @flow
import deprecationCheck from '../_deprecationCheck'

jest.mock('../_deprecated', () => ({
  'module/deprecatedModule.js': {
    version: '3.0',
    guidance: 'You should use the %c::currentModule%c instead.',
  },
}))

describe('deprecationCheck', () => {
  beforeAll(() => {
    global.console = {
      warn: jest.fn(),
      log: global.console.log,
    }
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  it('should throw a warning when the module is deprecated', () => {
    deprecationCheck('module/deprecatedModule.js')
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })

  it('should not throw a warning when the module is not deprecated', () => {
    deprecationCheck('module/currentModule.js')
    // eslint-disable-next-line no-console
    expect(console.warn.mock.calls).toMatchSnapshot()
    // eslint-disable-next-line no-console
    console.warn.mockClear()
  })
})
