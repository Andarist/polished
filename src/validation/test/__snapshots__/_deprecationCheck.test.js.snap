// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deprecationCheck should not throw a warning when the module is not deprecated 1`] = `Array []`;

exports[`deprecationCheck should throw a warning when the module is deprecated 1`] = `
Array [
  Array [
    "%c ✨ WARNING ✨ ---- module/deprecatedModule.js --%c

deprecatedModule %cwill be deprecated as of %cversion 3.0%c of ✨ polished. You should use the %c::currentModule%c instead.
  %c
Please see the documentation at %chttps://polished.js.org/docs/#deprecatedModule%c for more information.",
    "font-weight: bold; color: black",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: black; font-size: 12px; font-weight: bold",
    "color: black; font-size: 12px",
    "color: gray; line-height: 1.4",
    "color: blue; line-height: 1.4",
    "color: gray; line-height: 1.4",
  ],
]
`;
